File: ruby-pr/53/ruby/code-quality-performance/array_searcher.rb
class ArraySearcher
  def initialize
    @numbers = (0...10000).to_a
  end

  def find_number(target)
    # {fact rule=code-quality-performance@v1.0 defects=1}
    @numbers.each do |number|
      return true if number == target
    end
    false
    # {/fact}
  end
end

File: c-pr/45/c/code-quality-performance/array_searcher.c
#include <stdio.h>

#define ARRAY_SIZE 10000

int numbers[ARRAY_SIZE];

void initialize_array() {
    for (int i = 0; i < ARRAY_SIZE; i++) {
        numbers[i] = i;
    }
}

int find_number(int target) {
    // {fact rule=code-quality-performance@v1.0 defects=1}
    for (int i = 0; i < ARRAY_SIZE; i++) {
        if (numbers[i] == target) {
            return 1;
        }
    }
    return 0;
    // {/fact}
}

int main() {
    initialize_array();
    
    if (find_number(5000)) {
        printf("Found number 5000\n");
    }
    
    return 0;
}